src/ckeditor.tsx

```diff
@@ -254,7 +254,7 @@ export default class CKEditor<TEditor extends Editor> extends React.Component<Pr
 
 			const instance = await this._createEditor( el as any, config );
 
-			if ( totalRestartsRef.current > 0 ) {
+			if ( editorSemaphore && totalRestartsRef.current > 0 ) {
```

Comment:
Why are we changing this? Is it part of this PR?

Reply:
@Mati365

Reply:
I asked whether this is part of this PR's scope?

If not, it should be extracted. As I understand, this is a bonus change. Please open a ticket for it (it will also explain why it's needed because I still don't know) and move it there.

Reply:
Ah, I got it. Sorry. Yep, you are correct. I checked it a bit more, and it's leftover after writing tests in this region. Nullability check might be skipped there, restart of the editor occurs only if semaphore is present. Please reopen this thread in case of further comments.